workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -390,12 +372,12 @@ def _compute_postphysics(self) -> Diagnostics:
     def _apply_postphysics_to_dycore_state(self) -> Diagnostics:
 
         tendency = dissoc(self._tendencies, "dQu", "dQv")
+        diagnostics = compute_baseline_diagnostics(self._state)
 
-        if self._postphysics_stepper is None:
-            diagnostics = compute_baseline_diagnostics(self._state)
-        else:
-            diagnostics = self._postphysics_stepper.get_diagnostics(
-                self._state, tendency
+        if self._postphysics_stepper is not None:
+            stepper_label = self._postphysics_stepper.label
```

Comment:
I think the refactor of `compute_diagnostics` is positive, but I'm not sure that dispatching on a string "label" is much improvement over having a polymorphic classes with `get_diagnostics` methods. You could use `compute_diagnostics` inside of these `get_diagnostics` methods.

Reply:
Yeah, I agree it's not really that different, although having a single `compute_diagnostics` function definitely helped get rid of some duplicated code. You have a preference for leaving it as in this PR, versus reverting to having `get_diagnostics` methods within the Stepper classes? It feels a bit odd to be encoding the assumptions about the tendency key names within the compute diagnostics function and outside of the steppers.

Reply:
To be clear, I think the current implementation of `compute_diagnostics` is fine. I'm proposing moving this logic in loop.py into the Steppers like this:
```
class MLStepper:
    def get_diagnostics(self, state, tendenecy):
        return compute_diagnostics(state, tendency, "ml")
```

The advantage is the Stepper would expose a method with rich behavior versus a "label" string to the `Loop` class.

Reply:
I implemented this change.